src/metabase/models/native_query_snippet.clj

```diff
@@ -84,25 +85,31 @@
   [_model-name _opts snippet]
   (-> (serdes/extract-one-basics "NativeQuerySnippet" snippet)
       (update :creator_id serdes/*export-user*)
-      (update :collection_id #(when % (serdes/*export-fk* % 'Collection)))))
+      (m/update-existing :collection_id #(serdes/*export-fk* % 'Collection))))
 
 (defmethod serdes/load-xform "NativeQuerySnippet" [snippet]
   (-> snippet
       serdes/load-xform-basics
       (update :creator_id serdes/*import-user*)
-      (update :collection_id #(when % (serdes/*import-fk* % 'Collection)))))
+      (m/update-existing :collection_id #(serdes/*import-fk* % 'Collection))))
 
 (defmethod serdes/dependencies "NativeQuerySnippet"
   [{:keys [collection_id]}]
   (if collection_id
     [[{:model "Collection" :id collection_id}]]
     []))
 
-(defmethod serdes/storage-path "NativeQuerySnippet" [snippet ctx]
-  ;; Intended path here is ["snippets" "nested" "collections" "snippet_eid_and_slug"]
-  ;; We just the default path, then pull it apart.
-  ;; The default is ["collections" "nested" collections" "nativequerysnippets" "base_name"]
-  (let [basis  (serdes/storage-default-collection-path snippet ctx)
-        file   (last basis)
-        colls  (->> basis rest (drop-last 2))] ; Drops the "collections" at the start, and the last two.
-    (concat ["snippets"] colls [file])))
+(defmethod serdes/storage-path "NativeQuerySnippet" [snippet _ctx]
+  (let [{:keys [id label]} (-> snippet serdes/path last)]
+    ["snippets" (serdes/storage-leaf-file-name id label)]))
+
+(defmethod serdes/load-one! "NativeQuerySnippet" [ingested maybe-local]
+  (if (and (not (and maybe-local
+                     (= (:name ingested) (:name maybe-local))))
```

Comment:
why do we need this check? 

this works fine and pass your tests

```clojure
(if (t2/exists? :model/NativeQuerySnippet
                  :name (:name ingested) :entity_id [:!= (:entity_id ingested)])
    (serdes/default-load-one! (update ingested :name #(str % "-2")) maybe-local)
    (serdes/default-load-one! ingested maybe-local))
```

Reply:
Yeah, but it saves another query to a db. If we fetched a snippet by entity_id and the name is the same as incoming snippet's name, there is no need to do another check in db. Complicates code a little bit, but it's still short and very local, so IMO makes sense.

I'll add a comment about it.